apprentice_teacher.rb

```diff
@@ -1,45 +1,14 @@
-class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
-
-  def initialize(options={})
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
-    @target_raise = 800
-    @phase = 3
-  end
-
-  def offer_high_five
-    "High five!"
-  end
-
-  def set_phase(num)
-    @phase = num
-    "Cool, I've always wanted to teach phase #{num}!"
-  end
-
-  def teach_stuff
-    response = ""
-    response += "Listen, class, this is how everything works. "
-    response += "*drops crazy knowledge bomb* "
-    response += "... You're welcome."
-    response
-  end
-
-  def salary=(new_salary)
-    puts "This better be good!"
-    @salary = new_salary
-  end
-
-  def receive_raise(raise)
-    @salary += raise
-  end
+require_relative 'professor'
 
+class ApprenticeTeacher < Professor
+ 
+  TARGER_RAISE = 800
+ 
   def set_performance_rating(rating)
```

Comment:
This method can be abstracted out into the super class to be DRY. Any ideas on how to refactor the conditional and the ``` receive_raise ``` method call so it can work in both teacher and the apprentice?